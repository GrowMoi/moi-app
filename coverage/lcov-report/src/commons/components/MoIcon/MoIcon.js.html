<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/commons/components/MoIcon/MoIcon.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">src/commons/components/MoIcon</a> MoIcon.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">94.74% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>18/19</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">64.29% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>9/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">94.74% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>18/19</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';
import { Image, TouchableOpacity } from 'react-native';
import MoImages from './IconImages';
&nbsp;
const aspectRatio = (17 / 20);
&nbsp;
const IconImage = styled(Image)`
  width: ${props =&gt; props.size};
  height: ${props =&gt; (props.type === 'landscape' ? aspectRatio * props.size : props.size)};
`;
&nbsp;
export default class MoIcon extends Component {
  get sourceIcon() {
    const { name, active } = this.props;
    const moiCons = Object.keys(MoImages);
&nbsp;
    const nameActive = `${name}_active`;
    const hasActiveState = moiCons.includes(nameActive);
    const nameExist = moiCons.includes(name);
&nbsp;
    <span class="missing-if-branch" title="if path not taken" >I</span>if (!nameExist) {
<span class="cstat-no" title="statement not covered" >      return {</span>
        icon: MoImages.question.icon,
        type: MoImages.question.type,
      };
    }
&nbsp;
    return {
      icon: active &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >hasActiveState </span>? <span class="branch-0 cbranch-no" title="branch not covered" >MoImages[nameActive].icon </span>: MoImages[name].icon,
      type: active &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >hasActiveState </span>? <span class="branch-0 cbranch-no" title="branch not covered" >MoImages[nameActive].type </span>: MoImages[name].type,
    };
  }
&nbsp;
  render() {
    const { name, size, active, source, fadeDuration, resizeMode, onPress, ...rest } = this.props;
&nbsp;
    const iconImage = (
      &lt;IconImage
        source={this.sourceIcon.icon}
        size={size}
        resizeMode={resizeMode}
        type={this.sourceIcon.type}
        fadeDuration={fadeDuration}
        {...rest}
      /&gt;
    );
&nbsp;
    if (onPress) {
      return (
        &lt;TouchableOpacity onPress={onPress}&gt;
          {iconImage}
        &lt;/TouchableOpacity&gt;
      );
    }
&nbsp;
    return iconImage;
  }
}
&nbsp;
MoIcon.defaultProps = {
  size: 60,
  name: 'task',
  fadeDuration: 0,
  resizeMode: 'contain',
};
&nbsp;
MoIcon.propTypes = {
  name: PropTypes.string,
  size: PropTypes.number,
  active: PropTypes.bool,
  fadeDuration: PropTypes.number,
  source: PropTypes.any,
  resizeMode: PropTypes.string,
  onPress: PropTypes.func,
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Aug 04 2017 08:15:06 GMT-0500 (-05)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
