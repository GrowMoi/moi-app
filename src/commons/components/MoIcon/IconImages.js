export default {
  get question() {
    return {
      icon: require('../../../../assets/icons/question_mark.png'),
      type: 'square',
    };
  },

  get task() {
    return {
      icon: require('../../../../assets/icons/task/task.png'),
      type: 'square',
    };
  },
  get task_active() {
    return {
      icon: require('../../../../assets/icons/task/task_active.png'),
      type: 'square',
    };
  },

  get random() {
    return {
      icon: require('../../../../assets/icons/random/random.png'),
      type: 'square',
    };
  },
  get random_active() {
    return {
      icon: require('../../../../assets/icons/random/random_active.png'),
      type: 'square',
    };
  },

  get search() {
    return {
      icon: require('../../../../assets/icons/search/search.png'),
      type: 'landscape',
    };
  },
  get search_active() {
    return {
      icon: require('../../../../assets/icons/search/search_active.png'),
      type: 'landscape',
    };
  },

  get fav() {
    return {
      icon: require('../../../../assets/icons/fav/fav.png'),
      type: 'square',
    };
  },

  get profile() {
    return {
      icon: require('../../../../assets/icons/profile/profile.png'),
      type: 'square',
    };
  },

};
