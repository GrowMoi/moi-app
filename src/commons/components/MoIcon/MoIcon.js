import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';
import { Image, TouchableOpacity, View } from 'react-native';
import MoImages from './IconImages';
import Badge from '../Badge/Badge';

const aspectRatio = (17 / 20);
const aspectRatioBadge = 2.2;

const IconContainer = styled(View)`
  position: relative;
`;

const IconImage = styled(Image)`
  width: ${props => props.size};
  height: ${props => (props.type === 'landscape' ? aspectRatio * props.size : props.size)};
`;

const BadgeContainer = styled(View)`
  position: absolute;
  top: 0;
  right: -7;
`;

export default class MoIcon extends Component {
  get sourceIcon() {
    const { name, active } = this.props;
    const moiCons = Object.keys(MoImages);

    const nameActive = `${name}_active`;
    const hasActiveState = moiCons.includes(nameActive);
    const nameExist = moiCons.includes(name);

    if (!nameExist) {
      return {
        icon: MoImages.question.icon,
        type: MoImages.question.type,
      };
    }

    return {
      icon: active && hasActiveState ? MoImages[nameActive].icon : MoImages[name].icon,
      type: active && hasActiveState ? MoImages[nameActive].type : MoImages[name].type,
    };
  }

  render() {
    const {
      name,
      size,
      active,
      source,
      fadeDuration,
      resizeMode,
      onPress,
      badgeValue,
      ...rest
    } = this.props;

    const iconImage = (
      <IconContainer>
        <IconImage
          source={this.sourceIcon.icon}
          size={size}
          resizeMode={resizeMode}
          type={this.sourceIcon.type}
          fadeDuration={fadeDuration}
          {...rest}
        />
        {badgeValue > 0 && (
          <BadgeContainer>
            <Badge value={badgeValue} size={size / aspectRatioBadge} />
          </BadgeContainer>
        )}
      </IconContainer>
    );

    if (onPress) {
      return (
        <TouchableOpacity onPress={onPress}>
          {iconImage}
        </TouchableOpacity>
      );
    }

    return iconImage;
  }
}

MoIcon.defaultProps = {
  size: 60,
  name: 'task',
  fadeDuration: 0,
  resizeMode: 'contain',
  badgeValue: 0,
};

MoIcon.propTypes = {
  name: PropTypes.string,
  size: PropTypes.number,
  active: PropTypes.bool,
  fadeDuration: PropTypes.number,
  source: PropTypes.any,
  resizeMode: PropTypes.string,
  onPress: PropTypes.func,
  badgeValue: PropTypes.number,
};
