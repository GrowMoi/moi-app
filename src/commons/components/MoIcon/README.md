### Current Icons List
- question
- task - active
- random - active
- search - active
