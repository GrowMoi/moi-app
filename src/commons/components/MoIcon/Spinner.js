import React, { Component } from 'react';
import { Animated, Easing } from 'react-native';
import { FontAwesome } from '@expo/vector-icons';
import styled from 'styled-components/native';

import { Palette } from '../../styles';

const StyledIcon = styled(FontAwesome)`
  background-color: transparent;
`;
const Icon = Animated.createAnimatedComponent(StyledIcon);

export default class Spinner extends Component {
  spinValue = new Animated.Value(0);

  componentDidMount() {
    this.animate();
  }

  componentWillUnmount() {
    this.animation.stop();
  }

  get animation() {
    const { duration = 2000 } = this.props;

    return (
      Animated.timing(
        this.spinValue,
        {
          toValue: 1,
          duration: duration,
          easing: Easing.linear,
          useNativeDriver: true
        }
      )
    );
  }

  animate = () => {
    this.animation.start(() => this.animate());
  }

  render() {
    const { size = 18, color = Palette.dark.css(), name = 'spinner' } = this.props;
    const spin = this.spinValue.interpolate({
      inputRange: [0, 1],
      outputRange: ["0deg", "360deg"]
    });

    return (
      <Icon
        style={{transform: [{ 'rotate': spin }]}}
        name={name}
        size={size}
        color={color}
      />
    )
  }
}
