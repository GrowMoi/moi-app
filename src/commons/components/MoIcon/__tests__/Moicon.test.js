import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import MoIcon from '../MoIcon';

describe('Moicon', () => {
  it('it should render correctly with default props', () => {
    const wrapper = shallow(<MoIcon name="search" />);
    const tree = wrapper.dive();

    expect(tree).toMatchSnapshot();
    expect(tree).toBeTruthy();
  });

  it('it should render without name prop', () => {
    const wrapper = renderer.create(<MoIcon />);
    wrapper.toJSON();

    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toBeTruthy();
  });

  it('it should render Touchable structure with onPress prop exist', () => {
    const fn = jest.fn();
    const wrapper = renderer.create(<MoIcon name='fav' onPress={fn} />);
    wrapper.toJSON();

    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toBeTruthy();
  });
});
