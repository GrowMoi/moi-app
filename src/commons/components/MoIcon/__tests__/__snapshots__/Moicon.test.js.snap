// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Moicon it should render Touchable structure with onPress prop exist 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <Image
      fadeDuration={0}
      resizeMode="contain"
      size={60}
      source={1}
      style={
        Array [
          Object {
            "height": 60,
            "width": 60,
          },
          undefined,
        ]
      }
      type="square"
    />
  </View>
</View>
`;

exports[`Moicon it should render correctly with default props 1`] = `
<View
  style={
    Array [
      Object {
        "position": "relative",
      },
      undefined,
    ]
  }
>
  <Styled(Image)
    fadeDuration={0}
    resizeMode="contain"
    size={60}
    source={1}
    type="landscape"
  />
</View>
`;

exports[`Moicon it should render without name prop 1`] = `
<View
  style={
    Array [
      Object {
        "position": "relative",
      },
      undefined,
    ]
  }
>
  <Image
    fadeDuration={0}
    resizeMode="contain"
    size={60}
    source={1}
    style={
      Array [
        Object {
          "height": 60,
          "width": 60,
        },
        undefined,
      ]
    }
    type="square"
  />
</View>
`;
